src/lib/operators/jit_operator/jit_operations.hpp

```diff
@@ -0,0 +1,195 @@
+const auto jit_division = [](const auto& a, const auto& b) -> decltype(a / b) { return a / b; };
+const auto jit_modulo = [](const auto& a, const auto& b) -> decltype(a % b) { return a % b; };
+const auto jit_power = [](const auto& a, const auto& b) -> decltype(std::pow(a, b)) { return std::pow(a, b); };
+
+/* Comparison operators */
+const auto jit_equals = [](const auto& a, const auto& b) -> decltype(a == b, uint8_t()) { return a == b; };
+const auto jit_not_equals = [](const auto& a, const auto& b) -> decltype(a != b, uint8_t()) { return a != b; };
+const auto jit_less_than = [](const auto& a, const auto& b) -> decltype(a < b, uint8_t()) { return a < b; };
+const auto jit_less_than_equals = [](const auto& a, const auto& b) -> decltype(a <= b, uint8_t()) { return a <= b; };
+const auto jit_greater_than = [](const auto& a, const auto& b) -> decltype(a > b, uint8_t()) { return a > b; };
+const auto jit_greater_than_equals = [](const auto& a, const auto& b) -> decltype(a >= b, uint8_t()) { return a >= b; };
+
+const auto jit_like = [](const std::string& a, const std::string& b) -> uint8_t {
+  const auto regex_string = LikeTableScanImpl::sqllike_to_regex(b);
+  const auto regex = std::regex{regex_string, std::regex_constants::icase};
+  return std::regex_match(a, regex);
+};
+
+const auto jit_not_like = [](const std::string& a, const std::string& b) -> uint8_t {
+  const auto regex_string = LikeTableScanImpl::sqllike_to_regex(b);
+  const auto regex = std::regex{regex_string, std::regex_constants::icase};
+  return !std::regex_match(a, regex);
+};
+
+// The InvalidTypeCatcher acts as a fallback inplementation, if template specialization
+// fails for a type combination.
+template <typename F, typename R>
+struct InvalidTypeCatcher : F {
+  explicit InvalidTypeCatcher(F f) : F(std::move(f)) {}
+
+  using F::operator();
+
+  template <typename... Ts>
+  R operator()(const Ts...) const {
+    Fail("invalid combination of types for operation");
+  }
+};
+
+template <typename T>
+__attribute__((noinline)) void jit_compute(const T& op_func, const JitMaterializedValue& lhs,
+                                           const JitMaterializedValue& rhs, JitMaterializedValue& result) {
+  // Handle NULL values and return if either input is NULL.
+  result.is_null() = lhs.is_null() || rhs.is_null();
+  if (result.is_null()) {
+    return;
+  }
+
+  const auto func = [&](const auto& typed_lhs, const auto& typed_rhs, auto& result) -> decltype(
+      op_func(typed_lhs, typed_rhs), void()) {
+    using ResultType = decltype(op_func(typed_lhs, typed_rhs));
+    result.template as<ResultType>() = op_func(typed_lhs, typed_rhs);
+  };
+
+  const auto catching_func = InvalidTypeCatcher<decltype(func), void>(func);
+
+  // The type information from the lhs and rhs are combined into a single value for dispatching without nesting.
+  const auto combined_types = static_cast<uint8_t>(lhs.data_type()) << 8 | static_cast<uint8_t>(rhs.data_type());
```

Comment:
Mmmh, hacky but clever. Do we have a compile time check somewhere that there not more than $2^8$ data types?

Reply:
The `enum class` derives from `uint8_t`. [see here](https://github.com/hyrise/hyrise/blob/master/src/lib/all_type_variant.hpp#L52). That should take care of it.